shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/security/AlgorithmSecure.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.infra.security;
+
+import org.apache.shardingsphere.infra.spi.typed.TypedSPI;
+
+/**
+ * Typed of Secure.
+ */
+public interface AlgorithmSecure extends TypedSPI {
+    
+    /**
+     * encrypt input.
+     *
+     * @param input the input password
+     * @param charsetName the password' charset
+     * @return String the encrypted password
+     * @throws Exception when encrypt failed throw this exception
+     */
+    default String encrypt(final String input, final String charsetName) throws Exception {
+        return encrypt(input.getBytes(charsetName));
+    }
+
+    /**
+     * encrypt input.
+     *
+     * @param input the input password
+     * @return String the encrypted password
+     * @throws Exception when encrypt failed throw this exception
+     */
+    default String encrypt(final String input) throws Exception {
+        return encrypt(input, "UTF-8");
+    }
+
+    /**
+     * encrypt input.
+     *
+     * @param content the password byte array
+     * @return String the encrypted password
+     * @throws Exception when encrypt failed throw this exception
+     */
+    String encrypt(byte[] content) throws Exception;
```

Comment:
Is it necessary for encrypt with byte[] in password?

Reply:
yes, many encrypt algorithm input is byte[], etc: [0x1, 0x1, 0x3] ,this can't format with as String!
so in low level, this interface is need.